pkg/loghttp/push/otlp_test.go

```diff
@@ -0,0 +1,444 @@
+										Value: "bar - 1",
+									},
+									{
+										Name:  "resource_nested_foo",
+										Value: "bar",
+									},
+									{
+										Name:  "scope_nested_foo",
+										Value: "bar",
+									},
+									{
+										Name:  "scope_name",
+										Value: "fizz",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedStats: Stats{
+				numLines: 2,
+				logLinesBytes: map[time.Duration]int64{
+					time.Hour: 26,
+				},
+				structuredMetadataBytes: map[time.Duration]int64{
+					time.Hour: 97,
+				},
+				streamLabelsSize:         21,
+				mostRecentEntryTimestamp: now,
+			},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			stats := newPushStats()
+			pushReq := otlpToLokiPushRequest(tc.generateLogs(), "foo", fakeRetention{}, stats)
+			require.Equal(t, tc.expectedPushRequest, *pushReq)
+			require.Equal(t, tc.expectedStats, *stats)
+		})
+	}
+}
+
+func TestOTLPLogToPushEntry(t *testing.T) {
+	now := time.Unix(0, time.Now().UnixNano())
+
+	for _, tc := range []struct {
+		name           string
+		buildLogRecord func() plog.LogRecord
+		expectedResp   push.Entry
+	}{
+		{
+			name: "only body and timestamp set",
+			buildLogRecord: func() plog.LogRecord {
+				log := plog.NewLogRecord()
+				log.Body().SetStr("log body")
+				log.SetTimestamp(pcommon.Timestamp(now.UnixNano()))
+				return log
+			},
+			expectedResp: push.Entry{
+				Timestamp:          now,
+				Line:               "log body",
+				StructuredMetadata: push.LabelsAdapter{},
+			},
+		},
+		{
+			name: "all the values set",
+			buildLogRecord: func() plog.LogRecord {
+				log := plog.NewLogRecord()
+				log.Body().SetStr("log body")
+				log.SetTimestamp(pcommon.Timestamp(now.UnixNano()))
+				log.SetObservedTimestamp(pcommon.Timestamp(now.UnixNano() + 1))
+				log.SetSeverityNumber(plog.SeverityNumberDebug)
+				log.SetSeverityText("debug")
+				log.SetDroppedAttributesCount(1)
+				log.SetFlags(plog.DefaultLogRecordFlags.WithIsSampled(true))
+				log.SetTraceID([16]byte{0x12, 0x34, 0x56, 0x78, 0x12, 0x34, 0x56, 0x78, 0x12, 0x34, 0x56, 0x78, 0x12, 0x34, 0x56, 0x78})
+				log.SetSpanID([8]byte{0x12, 0x23, 0xAD, 0x12, 0x23, 0xAD, 0x12, 0x23})
+				log.Attributes().PutStr("foo", "bar")
+
+				return log
+			},
+			expectedResp: push.Entry{
+				Timestamp: now,
+				Line:      "log body",
+				StructuredMetadata: push.LabelsAdapter{
+					{
+						Name:  "foo",
+						Value: "bar",
+					},
+					{
+						Name:  "observed_timestamp",
+						Value: fmt.Sprintf("%d", now.UnixNano()+1),
+					},
+					{
+						Name:  "severity_number",
+						Value: "5",
+					},
+					{
+						Name:  "severity_text",
+						Value: "debug",
+					},
+					{
+						Name:  "dropped_attributes_count",
+						Value: "1",
+					},
+					{
+						Name:  "flags",
+						Value: fmt.Sprintf("%d", plog.DefaultLogRecordFlags.WithIsSampled(true)),
+					},
+					{
+						Name:  "trace_id",
+						Value: "12345678123456781234567812345678",
```

Comment:
It's a bit difficult to see how the traceID set in the payload maps to this output. Maybe we can have the traceID as a variable and call traceID.String() here?

Reply:
I am not sure if that would be an ideal test. I actually copied the test case from [otel repo](https://github.com/open-telemetry/opentelemetry-collector/blob/287b98f6973fd6baa278150b9fca8c83abea0af4/pdata/pcommon/traceid_test.go#L29).